components/browser/session-storage/src/main/java/mozilla/components/browser/session/storage/serialize/BrowserStateReader.kt

```diff
@@ -198,6 +204,9 @@ private fun JsonReader.tabSession(): RecoverableTab? {
             Keys.SESSION_LAST_MEDIA_URL -> lastMediaUrl = nextString()
             Keys.SESSION_LAST_MEDIA_SESSION_ACTIVE -> mediaSessionActive = nextBoolean()
             Keys.SESSION_LAST_MEDIA_ACCESS -> lastMediaAccess = nextLong()
+            Keys.SESSION_EXTERNAL_SOURCE_ID -> externalSourceId = nextInt()
+            Keys.SESSION_EXTERNAL_SOURCE_PACKAGE_ID -> externalSourcePackageId = nextStringOrNull()
+            Keys.SESSION_EXTERNAL_SOURCE_PACKAGE_CATEGORY -> externalSourceCategory = nextIntOrNull()
```

Comment:
This is a bit dangerous and we need to check the existing behaviors: We did not restore the source on purpose. There have been some parts of the code that look at new tabs with specific sources and log telemetry etc. If we now restore them they may show up as things like "new external app" to observers.

Reply:
A safer approach may be to introduce a new property - and potentially deprecate the existing one.

Reply:
@pocmo yeah we were concerned about this too, but I couldn't find a case where this would be a problem? Possible I missed it. Do you have any hints where to look? :)

Reply:
I was basically looking for readers of `SessionState.Source` which make decisions based on it, but didn't find much other than what's covered in: https://github.com/mozilla-mobile/fenix/pull/20578/files?

Reply:
OK made another pass and could find anything except: https://github.com/mozilla-mobile/fenix/pull/20578/files#r682729976 which we could handle, but not sure if worth it?

Reply:
OK, I think I would vote for going ahead and maybe handling the one case. Keeping both the old deprecated source and new one might cause other issues and extra work. Since we don't really know a case that could break maybe we should find out and worst case back out?

Reply:
Let's double check though in case I missed anything :).

Reply:
@pocmo we found the old ticket that caused us to not restore the source field: https://github.com/mozilla-mobile/fenix/issues/9937

Basically to make sure the back button behaviour is accurate for tabs originally opened from a VIEW intent where back should close the app, but when restored it shouldn't. We can introduce a separate state for this because technically `restored` is not a source.